zebra-chain/src/block/tests.rs

```diff
@@ -130,6 +136,160 @@ fn deserialize_block() {
         .expect("block test vector should deserialize");
 }
 
+#[test]
+fn block_limits_multi_tx() {
+    // Test multiple small transactions to fill a block max size
+    // A dummy transaction
+    let tx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]).unwrap();
+
+    // A block header
+    //let blockheader = generate_block_header();
+    let blockheader = zebra_test::generate::block_header().clone();
+
+    // Serialize header
+    let mut data_header = Vec::new();
+    blockheader
+        .zcash_serialize(&mut data_header)
+        .expect("Block header should serialize");
+
+    // Calculate the number of transactions we need
+    let mut max_transactions_in_block =
+        (MAX_BLOCK_BYTES as usize - data_header.len()) / *&zebra_test::vectors::DUMMY_TX1[..].len();
+
+    // Create transactions to be just below the limit
+    let mut many_transactions = std::iter::repeat(Arc::new(tx.clone()))
+        .take(max_transactions_in_block)
+        .collect::<Vec<_>>();
+
+    // Add the transactions into a block
+    let mut block = Block {
+        header: blockheader,
+        transactions: many_transactions.clone(),
+    };
+
+    // Serialize the block
+    let mut data = Vec::new();
+    block
+        .zcash_serialize(&mut data)
+        .expect("block should serialize as we are not limiting generation yet");
+
+    // Deserialize by now is ok as we are lower than the limit
+    let block2 = Block::zcash_deserialize(&data[..])
+        .expect("block should deserialize as we are just below limit");
+
+    assert_eq!(block, block2);
+    assert_eq!(max_transactions_in_block, block2.transactions.len());
+
+    // Add 1 more transaction to the block, limit will be reached
+    max_transactions_in_block = max_transactions_in_block + 1;
+    many_transactions.push(Arc::new(tx));
+    block.transactions = many_transactions.clone();
+    assert_eq!(max_transactions_in_block, many_transactions.len());
```

Comment:
Suggestion: Can you move this oversize block generation code to `oversize_multi_transaction_block()`  in a `zebra_chain::test::generate` module?